statement.k

```diff
@@ -185,10 +185,92 @@ imports CONFIGURATION
          rule if (false) S:Statement => .
          rule if (true)  S else S1:Statement => S
          rule if (false) S1:Statement else S => S
-
+/*
          rule for (A;B;C) S => A; while (B) {S C;} [structural]
          rule do S while (B) =>  S if(B) {do S while (B)} else {.Statements}  [structural]
-         rule while (B) S => if (B) {S while (B) S} else {.Statements}  [structural]
+         rule while (B) S => if (B) {S while (B) S} else {.Statements}  [structural]*/
+
+          rule <k> for (A;B;C) {S} => InitStmts(A;) if (B) {ExeWhile(C; S) for (A;B;C) {S}} else {EndLoop(.Statements)} ... </k>
+              <loopCon> true </loopCon>
+              <Continue> _=>false </Continue>
+              <funCond> true </funCond>
+
+         rule <k> for (A;B;C) {S} => Reset(1) ... </k> 
+              <loopCon> false </loopCon>
+
+         rule <k> for (A;B;C) {S} => . ... </k> 
+              <funCond> false </funCond>
+
+         rule <k> EndLoop(.Statements) => . ... </k>
+              <forInit> _=>true </forInit>
+
+         rule <k> InitStmts(S:Statement Ss:Statements) => InitStmt(S) ~> InitStmts(Ss) ... </k>
+         rule <k> InitStmt(S) => S ... </k>
+              <forInit> true=>false </forInit>
+         rule <k> InitStmt(S) => . ... </k>
+              <forInit> false </forInit>
+         rule <k> InitStmts(.Statements) => . ... </k>
+     
+              
+            
+
+         rule <k> do {S} while (B) =>  ExeWhile(S) if(B) {do {S} while (B)} else {.Statements} ... </k> 
+              <loopCon> true </loopCon>
+              <Continue> _=>false </Continue>
+              <funCond> true </funCond>
+
+        rule <k> do {S} while (B) => Reset(1) ... </k> 
+             <loopCon> false </loopCon>
+
+        rule <k> do {S} while (B) => . ... </k> 
+             <funCond> false </funCond>
+
+         
+
+               
+         rule <k> while (B) {S} => if (B) {ExeWhile(S) while (B) {S}} else {.Statements} ... </k> 
+              <loopCon> true </loopCon>
+              <Continue> _=>false </Continue>
+              <funCond> true </funCond>
+             
+
+
+         rule <k> ExeWhile(S:Statement Ss:Statements) => ExeWhileStmt(S) ~>  ExeWhile(Ss) ... </k>
+         rule <k> ExeWhile(.Statements) => . ... </k>
+
+         rule <k> ExeWhileStmt(S) => S ... </k>
+              <funCond> true </funCond>
+              <Continue> false </Continue>
+              <loopCon> true </loopCon>
+
+         rule <k> ExeWhileStmt(S) => . ... </k>
+              <funCond> false </funCond>
+
+         rule <k> ExeWhileStmt(S) => . ... </k>
+             <Continue> true </Continue>
+
+         rule <k> ExeWhileStmt(S) => . ... </k>
+             <loopCon> false </loopCon>
```

Comment:
Do we really want this to be called `loopCon`? Or is it something else? Perhaps something like `shouldSkip`? 
Calling it `loopCon` sounds a bit too generic for me, plus there are some other things around which can make it confusing (e.g. `funConf` etc.).